### PR TITLE
[pi-pir-sensor] Added types for pir-pir-sensor package

### PR DESCRIPTION
**Pull request by mariomurrent-softwaresolutions**
If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/pi-pir-sensor/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+declare function Sensor(configuration: Sensor.SensorConfiguration): void;
+
+declare namespace Sensor {
+    class Sensor {
+        lastMovement: Date;
+
+        constructor(sensorConfiguration: SensorConfiguration);
+
+        start(): void;
```
Comment (peterblazejewicz) [inline types/pi-pir-sensor/index.d.ts:16]: this one IMO accepts callback `(err) => void`:
https://github.com/opedromiranda/pi-pir-sensor/blob/master/lib/sensor.js#L30

File: types/pi-pir-sensor/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+        stop(): void;
```
Comment (peterblazejewicz) [inline types/pi-pir-sensor/index.d.ts:18]:
```suggestion
        stop(): boolean;
```
https://github.com/opedromiranda/pi-pir-sensor/blob/0b8d7d2bd2d153883c6c4e88fa55cacd4f396db3/lib/sensor.js#L49-L58

**Comment by peterblazejewicz** [comment]
it looks this module exports class construct directly (th'ats old module, so a function with class function). So more like:
```ts
declare class Sensor {
    lastMovement: Date;

    constructor(sensorConfiguration: Sensor.SensorConfiguration);

    start(cb?: (err: Error) => void): void;

    stop(): boolean;

    on(event: string, callback: () => void): void;
}

declare namespace Sensor {
    interface SensorConfiguration {
        pin: number;
        loop?: number;
    }
}

export = Sensor;
```
```ts
import * as Sensor from 'pi-pir-sensor';

// With default loop of 1500
const sensor: Sensor = new Sensor({pin: 12});
```

**Comment by mariomurrent-softwaresolutions** [comment]
Thanks for the review, I will check this tomorrow 😊

**Comment by peterblazejewicz** [comment]
are you really able to create an instance of the sensor?:
```text
TypeError: pirSensor.Sensor is not a constructor
    at Object.<anonymous> (C:\develop\tests\index.js:4:16)
```
please read my comment. It seems module exports class directly (that is a Function constructor)

**Comment by mariomurrent-softwaresolutions** [comment]
@peterblazejewicz hmm, that's strange. I'm not getting this error in my IDE

**Comment by mariomurrent-softwaresolutions** [comment]
@peterblazejewicz sorry, I have changed the implementation according to your suggestions. Don't know why the test fails locally but work here.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@mariomurrent-softwaresolutions thx!

**Comment by mariomurrent-softwaresolutions** [comment]
Ready to merge

**Comment by mariomurrent-softwaresolutions** [comment]
> LGTM!
> 
> @mariomurrent-softwaresolutions thx!

Thanks for your time and reviewing 😊

**Pull request merged**